content/backend/graphql-scala/6-interfaces.md

```diff
@@ -0,0 +1,352 @@
+1. Add `Vote` class with fields: `id`, `createdAt`, `userId`, `linkId`(you no need to define any relations for now)
+1. Create database tables for boths,
+1. Add object types for boths,
+1. Add fetchers for both,
+1. Implement `HasId` type class,
+1. Add fields in main `ObjectType` which allows for fetch list of entities like `users` and `votes`
+
+Try it...
+
+### User entity
+
+Let's start from user entity:
+
+<Instruction>
+Add `User.scala` class with content:
+
+```scala
+case class User(id: Int, name: String, email: String, password: String, createdAt: DateTime = DateTime.now)
+```
+
+</Instruction>
+
+Database setup.
+
+<Instruction>
+
+Add following content to the `DBSchema` class:
+
+```scala
+//val Links = TableQuery[LinksTable]
+
+class UsersTable(tag: Tag) extends Table[User](tag, "USERS"){
+  def id = column[Int]("ID", O.PrimaryKey)
+  def name = column[String]("NAME")
+  def email = column[String]("EMAIL")
+  def password = column[String]("PASSWORD")
+  def createdAt = column[DateTime]("CREATED_AT")
+
+  def * = (id, name, email, password, createdAt).mapTo[User]
+}
+
+val Users = TableQuery[UsersTable]
+```
+
+</Instruction>
+
+Sample entities:
+
+<Instruction>
+
+In file DBSchema in function `databaseSetup`: Add an action `Users.schema.create` at beginning of the sentence and then
+add few users later in this function
+
+```scala  
+Users ++= Seq(
+    User(1, "mario", "mario@example.com", "s3cr3t"),
+    User(2, "Fred", "fred@flinstones.com", "wilmalove")
+  )
+```
+
+</Instruction>
+
+Add user retrieval function:
+
+<Instruction>
+
+In `DAO` class add a function:
+
+```scala
+def getUsers(ids: Seq[Int]): Future[Seq[User]] = {
+    db.run(
+      Users.filter(_.id inSet ids).result
+    )
+}
+```  
+
+</Instruction>
+
+GraphQL part:
+
+<Instruction>
+
+In `GraphQLSchema` add :
+
+```scala
+implicit val UserType = deriveObjectType[Unit, User]() //ObjectType for user
+implicit val userHasId = HasId[User, Int](_.id) //HasId type class
+val usersFetcher = Fetcher(
+    (ctx: MyContext, ids: Seq[Int]) => ctx.dao.getUsers(ids)
+)// resolver
+
+```
+
+</Instruction>
+
+Add fetcher to resolvers.
+
+<Instruction>
+
+Add lastly created fetcher to the resolvers list. In the same file, replace constant `Resolver` with:
+
+```
+val Resolver = DeferredResolver.fetchers(linksFetcher, usersFetcher)
+```
+
+</Instruction>
+
+Add fields to main ObjectType:
+
+<Instruction>
+
+Add to `QueryType.fields`:
+
+```scala
+//val Ids = Argument("ids", ListInputType(IntType))
+Field("users",
+        ListType(UserType),
+        arguments = List(Ids),
+        resolve = c => usersFetcher.deferSeq(c.arg(Ids))
+)
+```
+
+</Instruction>
+
+We're ready.. you can now execute query like this:
+
+```graphql
+
+query {
+
+    users(ids: [1,2]){
+    	id
+			name
+    	email
+    	createdAt
+  	}
+}  
+```
+
+
+### Vote entity
+
+If you want, you can make similar step for `Vote` entity. And then follow instructions and check everything is ok.
```

Comment:
Why this paragraph? I think earlier we asked the user to work on both `user`s and `vote`s